feat(virtualization-hyperv-2012-cma): initial release CTOR-1832 (#4646)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/virtualization-hyperv-2012-cma.md

---
id: virtualization-hyperv-2012-cma
title: Hyper-V 2012 CMA
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Connector dependencies

The following monitoring connectors will be installed when you install the **Hyper-V 2012 CMA** connector through the

> **Configuration > Monitoring Connector Manager** menu:

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu:
```

| *vm*#dataexchange-status | N/A   |
| *vm*#heartbeat-status    | N/A   |
| *vm*#backup-status       | N/A   |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
<TabItem value="Scvmm-Snapshot" label="Scvmm-Snapshot">

| Metric name   | Unit  |
|:--------------|:------|
| *vm*#snapshot | N/A   |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
<TabItem value="Scvmm-Vm-Status" label="Scvmm-Vm-Status">

| Metric name | Unit  |
|:------------|:------|
| *vm*#status | N/A   |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
</Tabs>

## Prerequisites

### Network flow

Only one TCP flow must be open from the host to the poller.

| Source         | Destination | Protocol | Port | Purpose                                              |
|----------------|-------------|----------|------|------------------------------------------------------|
| Monitored host | Collecteur  | TCP      | 4317 | Configuration retrieval, and OpenTelemetry data flow |

### System prerequisites on the poller

> > To be able to use the Centreon Monitoring agent, you must use a poller with at least version <!--`24.09.0` for Centreon Cloud users and version--> `24.04.6` or `24.10.0` for On Prem users of `centreon-engine`. The Centreon Monitoring agent will configure itself by connecting to Centreon Engine.

Reviewer: that bit is not commented in FR I think

### Configure Engine

[Configure how the poller and the agent will communicate](../getting-started/how-to-guides/cma.md#configure-polleragent-communication).

### System prerequisites on the monitored host

> The installer can be downloaded from the [centreon-collect's releases page](https://github.com/centreon/centreon-collect/releases?q=centreon-collect&expanded=true).

Reviewer: same as FR

> The installer can be downloaded from the [centreon-collect's releases page](https://github.com/centreon/centreon-collect/releases?q=centreon-collect&expanded=true).

Author: ```suggestion
The installation and configuration procedure of Centreon Monitoring Agent for Windows is detailed in [this dedicated page](https://docs.centreon.com/pp/integrations/plugin-packs/getting-started/how-to-guides/cma/cma-setup/#step-3-prepare-the-host).
```

| CRITICALSTATUS    | Define the conditions to match for the status to be CRITICAL. You can use the following variables: %\{vm\}, %\{status\}, %\{hostgroup\} | not %\{status\} =~ /Running\|Stopped/i |           |
| WARNINGSTATUS     | Define the conditions to match for the status to be WARNING . You can use the following variables: %\{vm\}, %\{status\}, %\{hostgroup\} |                                        |           |
| EXTRAOPTIONS      | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options).  | --verbose                              |           |

</TabItem>
</Tabs>

3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.

## How to check in the CLI that the configuration is OK and what are the main options for?

> Once the plugin is installed, log into your Centreon poller's CLI using the

Reviewer: same as FR

> Once the plugin is installed, log into your Centreon poller's CLI using the

Author: ```suggestion
Once the plugin is installed, log into your host's CLI (this command cannot be run on the pollers) using the
```
